`/files` エンドポイントのGETメソッドに関する詳細設計を追加

a5ob7r opened the pull request:
# 概要

`/files` エンドポイントのGETメソッドに関する詳細設計をOpenAPI v3.1.0に沿って記述した。

エラーレスポンスのパラメータは、OAuth 2.0やProblem DetailsのRFCを参考に記述した。

# 関連知識

- OpenAPI Specification v3.1.0 - https://spec.openapis.org/oas/v3.1.0
- JSON Schema - https://json-schema.org
- OAuth Parameters - https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml

# 補足

本プルリクは #5 の後続である。 #5 のマージ後にベースブランチをmainに変更する。

a5ob7r committed:
Add an description for `files`'s get

tk0miya commented on openapi.yaml:
```diff
@@ -190,7 +194,100 @@ paths:
+                      }
+                    ]
+        '401':
+          description: ユーザがアップロードしたファイル一覧の取得に失敗したことを意味する。
```
この説明はちょっと情報量が足りないな、と感じました。
`error` カラムの内容を見た印象だと、一般的な 401 (認証が必要) を示すエラーのように見えます。
取得に失敗という説明は幅広いので、もう少し掘り下げたいですね。

※ insufficient_scope は権限不足なので 403 Forbidden の方が適当でしょうね

a5ob7r replied:
ドキュメントを読んでみたところ、各エラーコードに対応するHTTPステータスコードが記載されていました。

その仕様が合致するように、 [469c874](https://github.com/sls-training/2023-API-training-server/pull/6/commits/469c874efe64d91c3e663c960a5818941d084e43) でエラーレスポンスを適切な複数のHTTPステータスに分けて記述するよう修正しました。

tk0miya replied:
HTTP のステータスコードは頻出なので、かるく頭に入れておいてください (暗記ものとまでは言わないけれど、開発者の共通認識ではあるので番号での会話がよく行われます)

tk0miya reviewed (approved):
問題なさそう

tim-toga commented on openapi.yaml:
```diff
@@ -190,7 +194,100 @@ paths:
+                type: array
+                items:
+                  type: object
+                  properties:
```
一覧を取得するAPIは、page,perなどのパラメータで何件取得するかを指定できるようにした方が使いやすいですね。

似たようなので`/files/{fileId}`もあるけど、これは1件のみの取得なので。

a5ob7r replied:
あー、ページネーションみたいな感じですかね。

a5ob7r replied:
[8dc6a5d](https://github.com/sls-training/2023-API-training-server/pull/6/commits/8dc6a5df9ea0dfef787d41acf2edb24ed00aeeb1) にて、ページネーション関連のパラメータを追加しました。

tim-toga replied:
リンク先が存在しない。あと、パラメータが追加されてないように見える

a5ob7r replied:
おっと、こちらもリベースしたから元のコミットが消えてしまっていますね。こちらになります。
[8ad6345](https://github.com/sls-training/2023-API-training-server/pull/6/commits/8ad6345f12126587ac12de3c678235d5b29f9191)

tk0miya replied:
GitHub は結構賢い差分表示をしてくれるので、コミット ID を貼らなくてもいいと思うよ。
前回見た時点からの差分を表示、みたいなことができる。

※ こうした機能をうまく使うには、コミット ID が変わらない方がよいので、なるべく rebase は避けてもらったほうがありがたかったりします。

a5ob7r replied:
ひとまずリベースを使わないようにしてみます。

a5ob7r committed:
Split an error response into multiple them

Each of error codes should match the appropriate HTTP status code
according to the RFC.

https://datatracker.ietf.org/doc/html/rfc6750#section-3.1

a5ob7r committed:
Add some parameters for paginations

a5ob7r committed:
Add `id` attribute to `/files` endpoint's response

a5ob7r commented:
Approve を2つ得られたのでマージします。
